### PR TITLE
[refactor](planner)Cancel the return value default 65535 limit

### PR DESCRIPTION
**Pull request by FreeOnePlus**
# Proposed changes
Originally, Order By Limit returned a maximum of 65535 rows of data by default during the query, but now many businesses do not apply this limit. It is necessary to add larger data after the query statement to complete the full data query, which is extremely inconvenient, so adjustments have been made.

At the same time, I added the variable DEFAULT_ORDER_BY_LIMIT to the SessionVariable, the default value is -1, if the user does not use the LIMIT keyword or the LIMIT value is a negative integer, the default query return value is Long.MAX_VALUE. If the corresponding maximum query value is set, the number of data items is returned according to the maximum query value or the value followed by the LIMIT keyword.

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java [hunk 1]
```diff
@@ -232,7 +232,14 @@ private PlanNode createQueryPlan(QueryStmt stmt, Analyzer analyzer, long default
             throws UserException {
         long newDefaultOrderByLimit = defaultOrderByLimit;
         if (newDefaultOrderByLimit == -1) {
-            newDefaultOrderByLimit = 65535;
+            if (analyzer.getContext().getSessionVariable().defaultOrderByLimit <= -1) {
+                newDefaultOrderByLimit = Long.MAX_VALUE;
+                LOG.warn("The current default Order By Limit number is the maximum value of Long.");
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java:237]: remove the warn log

File: fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java [hunk 2]
```diff
@@ -232,7 +232,14 @@ private PlanNode createQueryPlan(QueryStmt stmt, Analyzer analyzer, long default
...
+            } else {
+                newDefaultOrderByLimit = analyzer.getContext().getSessionVariable().defaultOrderByLimit;
+                LOG.warn("The current default Order By Limit number is {}." + newDefaultOrderByLimit);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java:240]: remove warn log

File: fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java [hunk 3]
```diff
@@ -232,7 +232,14 @@ private PlanNode createQueryPlan(QueryStmt stmt, Analyzer analyzer, long default
             throws UserException {
         long newDefaultOrderByLimit = defaultOrderByLimit;
         if (newDefaultOrderByLimit == -1) {
-            newDefaultOrderByLimit = 65535;
+            if (analyzer.getContext().getSessionVariable().defaultOrderByLimit <= -1) {
+                newDefaultOrderByLimit = Long.MAX_VALUE;
+                LOG.warn("The current default Order By Limit number is the maximum value of Long.");
+            } else {
+                newDefaultOrderByLimit = analyzer.getContext().getSessionVariable().defaultOrderByLimit;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java:239]: you can just call `analyzer.getContext().getSessionVariable().defaultOrderByLimit` once

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 4]
```diff
@@ -230,6 +232,10 @@ public class SessionVariable implements Serializable, Writable {
     @VariableMgr.VarAttr(name = ENABLE_EXCHANGE_NODE_PARALLEL_MERGE)
     public boolean enableExchangeNodeParallelMerge = false;
 
+    // Default Order By Limit
+    @VariableMgr.VarAttr(name = DEFAULT_ORDER_BY_LIMIT)
+    public long defaultOrderByLimit = -1;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:237]: add document

File: fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java (hunk 2 above)
Comment (FreeOnePlus) [reply fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java]: has been completed

File: fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java (hunk 3 above)
Comment (FreeOnePlus) [reply fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java]: has been completed

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java (hunk 4 above)
Comment (FreeOnePlus) [reply fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:238]: has been completed

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.72 seconds
 load time: 432 seconds
 storage size: 17150866967 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221118043009_clickbench_pr_48491.html

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**